react/features/mobile/callkit/middleware.js

```diff
@@ -0,0 +1,194 @@
+/* @flow */
+
+import uuid from 'uuid';
+
+import {
+    APP_WILL_MOUNT,
+    APP_WILL_UNMOUNT,
+    appNavigate
+} from '../../app';
+import {
+    CONFERENCE_FAILED,
+    CONFERENCE_LEFT,
+    CONFERENCE_WILL_JOIN,
+    CONFERENCE_JOINED
+} from '../../base/conference';
+import { getInviteURL } from '../../base/connection';
+import {
+    SET_AUDIO_MUTED,
+    SET_VIDEO_MUTED,
+    isVideoMutedByAudioOnly,
+    setAudioMuted
+} from '../../base/media';
+import { MiddlewareRegistry, toState } from '../../base/redux';
+import { _SET_CALLKIT_LISTENERS } from './actionTypes';
+import CallKit from './CallKit';
+
+/**
+ * Middleware that captures several system actions and hooks up CallKit.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
+    const result = next(action);
+
+    switch (action.type) {
+    case _SET_CALLKIT_LISTENERS: {
+        const { listeners } = getState()['features/callkit'];
+
+        if (listeners) {
+            for (const [ event, listener ] of listeners) {
+                CallKit.removeEventListener(event, listener);
+            }
+        }
```

Comment:
The above's useless because the `case _SET_CALLKIT_LISTENERS` is executed after the action has reached the reducer thanks to first executing `next(result)`.